packages/generate/src/syntax/toEdgeQL.ts

```diff
@@ -932,10 +938,11 @@ function renderEdgeQL(
       expr.__element__.__name__ === "std::number"
         ? "std::float64"
         : expr.__element__.__name__;
-    if (expr.__expr__ === null) {
-      return `<${typeName}>{}`;
-    }
-    return `<${typeName}>(${renderEdgeQL(expr.__expr__, ctx)})`;
+    const inner = expr.__expr__
+      ? `(${renderEdgeQL(expr.__expr__, ctx)})`
+      : `{}`;
+    // TODO OPTION: just always wrap casts
+    return `(<${typeName}>${inner})`;
```

Comment:
Yeah, I think it's fine to unconditionally wrap casts in parens for now. Simplest, least error prone, and I don't find it to be too unreadable. This was going to be my approach!